src/roswire/common/format.py

```diff
@@ -30,85 +35,46 @@ class FormatDatabase:
         An immutable mapping from action name to definition.
     """
 
-    @staticmethod
-    def build(db: PackageDatabase) -> "FormatDatabase":
+    @classmethod
+    @abstractmethod
+    def build(cls, db: PackageDatabase) -> "FormatDatabase":
```

Comment:
Maybe change this to `from_packages(cls, db: PackageDatabase)` and add a `build(cls, messages, services, actions)` method.